modules/rostests/apitests/ntdll/NtQueryInformationProcess.c

```diff
@@ -323,6 +325,133 @@ Test_ProcessWx86Information(void)
     trace("VdmPower = %lu\n", VdmPower);
 }
 
+static
```

Comment:
Since most of this code seems to be identical, consider placing it into a library function:

```
void
QueryySetInformationAlignmentValidator(
    PINFORMATION_CLASS_INFO InformationClassInfo,
    PQUERY_SET_INFO_CALLBACK Callback)
{
    for (...)
    {
        Callback(InfoClass, UnalignedPointer, Size, ExpectedStatus);
        Callback(InfoClass, InvalidPointer, Size, ExpectedStatus);
        // ...
    }
}

void
QueryProcessInformationValidatorCallback(
    ULONG InfoClass,
    PVOID InfoPointer,
    ULONG InfoSize,
    NTSTATUS ExpectedStatus,
    ULONG ExpectedReturnedSize)
{
    NTSTATUS Status;
    ULONG ReturnedSize;

    switch (InfoClass)
    {
        case ProcessWorkingSetWatch:
            AlternativeStatus = STATUS_UNSUCCESSFUL;
         // ...
    }

    Status = NtQueryInformationProcess(NtCurrentProcess(),
                                           InfoClass,
                                           InfoPointer,
                                           InfoSize,
                                           &ReturnedSize);

    ok(Status == ExpectedStatus || Status == AlternativeStatus, "...");
    ok(ReturnedSize == ExpectedReturnedSize);
}
```